app/javascript/components/server-components/DownloadPage/PageListLayout.tsx

```diff
@@ -0,0 +1,58 @@
+import { Slot } from "@radix-ui/react-slot";
+import * as React from "react";
+
+import { classNames } from "$app/utils/classNames";
+
+export const PageListLayout = ({
+  pageList,
+  children,
+  className,
+}: {
+  pageList: React.ReactNode;
+  children: React.ReactNode;
+  className?: string;
+}) => (
+  <div
+    className={classNames(
+      "@container flex h-full flex-col gap-x-6 gap-y-16 overflow-y-auto bg-background p-4 [scrollbar-gutter:stable] md:p-8 lg:grid lg:grid-cols-[20rem_1fr]",
+      className,
+    )}
+  >
+    <div className="flex flex-col gap-4 lg:sticky lg:top-0 lg:overflow-y-auto lg:pr-2 lg:pb-8 lg:@max-[100vh]:max-h-[100cqh]">
+      {pageList}
+    </div>
+    {children}
+  </div>
+);
+
+export const PageList = React.forwardRef<HTMLDivElement, React.HTMLProps<HTMLDivElement>>(
```

Comment:
Do we actually use the ref anywhere?

Reply:
We need it for compatibility with ReactSortable in `app/javascript/components/ProductEdit/ContentTab/index.tsx`